src/libasr/pass/simplifier.cpp

```diff
@@ -2245,14 +2308,17 @@ class VerifySimplifierASROutput:
         return ;
     }
 
-    #define check_for_var_if_array(expr) if( is_temporary_needed(expr) ) { \
-            LCOMPILERS_ASSERT(ASR::is_a<ASR::Var_t>(*ASRUtils::get_past_array_physical_cast(expr))); \
-        } \
+    void check_for_var_if_array(ASR::expr_t* expr) {
+        if ( is_temporary_needed(expr) ) {
+            LCOMPILERS_ASSERT(ASR::is_a<ASR::Var_t>(*ASRUtils::get_past_array_physical_cast(expr)));
+        }
+    }
 
-    #define check_if_linked_to_target(expr, type) if( ASRUtils::is_aggregate_type(type) \
-        && ASRUtils::is_simd_array(type) ) { \
-         LCOMPILERS_ASSERT( exprs_with_target.find(&(const_cast<ASR::expr_t&>(expr))) != \
-                            exprs_with_target.end()); \
+    void check_if_linked_to_target(ASR::expr_t expr, ASR::ttype_t* type) {
+        if( ASRUtils::is_aggregate_type(type) &&
+            ASRUtils::is_simd_array(type) ) {
+            LCOMPILERS_ASSERT(exprs_with_target.find(&expr) != exprs_with_target.end());
+        }
```

Comment:
Ahhh, here is the catch, in release mode we do not consider the `LCOMPILERS_ASSERT` it is for Debug mode and the only place `expr` is used is `LCOMPILERS_ASSERT(exprs_with_target.find(&expr) != exprs_with_target.end());` hence the error.
```suggestion
    void check_if_linked_to_target([[maybe_unused]] ASR::expr_t expr, ASR::ttype_t* type) {
        if( ASRUtils::is_aggregate_type(type) &&
            ASRUtils::is_simd_array(type) ) {
            LCOMPILERS_ASSERT(exprs_with_target.find(&expr) != exprs_with_target.end());
        }
```